ports/esp32s2/common-hal/pulseio/PulseOut.c

```diff
@@ -82,5 +85,7 @@ void common_hal_pulseio_pulseout_send(pulseio_pulseout_obj_t* self, uint16_t* pu
     }
 
     rmt_write_items(self->channel, items, length, true);
-    rmt_wait_tx_done(self->channel, pdMS_TO_TICKS(100));
+    while (rmt_wait_tx_done(self->channel, 0) != ESP_OK) {
+        RUN_BACKGROUND_TASKS();
```

Comment:
```suggestion
        RUN_BACKGROUND_TASKS;
```

Sorry I got this wrong. No `()`. (CI is unhappy with it.)